API.md

```diff
@@ -503,36 +752,119 @@ frisby.create('First test')
 ```
 
 ### only()
+
 Exclusively run this test. When `toss()` is invoked, the test is wrapped in a
 Mocha `describe.only` block instead of a `describe` block.
 
+### timeout(ms)
+
+Sets the timeout for this request.
+
+- Types: `ms`: `string`
+- Defaults: `none`
+
+```javascript
+frisby.create('Long-running request')
+    .get('http://example.com/slowthing')
+    .timeout(5000)
+    .expectStatus(200)
+    .toss()
+```
+
+This function can also be called with no parameter to return the current configured timeout (either by default, by global setup or having used this function with a parameter previously).
+
+### retry(count, backoff)
+
+Set the number of (and additional backoff between) retries for this test. Each retry will be the configured timeout plus (retry number x backoff) apart.
+
+- Types: `count`: `integer`, `backoff`: `integer`
+- Defaults: `backoff`: 1000
+
+```javascript
+frisby.create('Get a flaky thing')
+    .get('http://example.com/thing-that-sometimes-responds')
+    .expectStatus(200)
+    .retry(2, 250)
+    .toss()
+```
+
+### baseUri(uri)
+
+Set the root URI/URL that will be prepended to every request, replacing anything set by `request.baseUri` in global setup.
+
+```javascript
+frisby.create('Simple Get')
+  .baseUri('http://httpbin.org')
+  .get('/get?foo=bar')
+  .expectStatus(200)
+.toss()
+```
+
+### waits(ms)
+
+Sets a period of time in milliseconds to wait after `toss()` before the request is sent
+
+```javascript
+const myUser = {name: 'Jane Doe'}
+frisby.create('Create Item')
+    .post('http://example.com/users/create', myUser)
+    .after(function(err, res, body, headers) {
+        frisby.create('Check Item Exists')
+            .get('http://example.com/users/list')
+            .waits(500) //Allow time for extra server-side processing of the user
+            .expectJSON('?', myUser)
+            .toss()
+    })
+    .toss()
+```
+
+### exceptionHandler(function)
+
+Sets a function to run if an error is raised. Can be used to output additional debug info not covered by the [inspectors](#inspectors), or perhaps to add validation to a non-deterministic result.
```

Comment:
Just curious, do you use this anywhere? It's a behavior I'm not sure we need. What you're saying about additional debug info makes sense, though it seems like we could do that in `inspect`. You mention validation for a non-determinstic result, and I'm not sure I understand that. I'm kind of puzzled.

Reply:
My only usage so far has been in IcedFrisby unit tests, where it's got lots of usage. That said, I can imagine a case where I want to assert that the response for "pending notifications", an endpoint that for "reasons" is immune to Arrange/Act/Assert, is either empty or matches a specific format. I could use exceptionHandler for this.

Not sure I entirely believe this use case fully myself (because if we wanted to support "either A or B" expects, we could do a much better job), but I definitely wanted to include something about this method as it's publicly available. An alternative could be renaming it to `_exceptionHandler`.

Reply:
Couldn't this use case be handled more cleanly using `.after()`?

> … I definitely wanted to include something about this method as it's publicly available. An alternative could be renaming it to `_exceptionHandler`.

I'm fine with including it for the moment, though like your idea of renaming it.